src/Export/Feed.php

```diff
@@ -16,22 +17,64 @@ class Feed
     /** @var FilterInterface */
     private $filter;
 
+    /** @var array */
+    private $currencyList;
+
+    /** @var CurrencyEntity */
+    private $defaultCurrency;
+
     public function __construct(DataProviderInterface $dataProvider, FilterInterface $filter)
     {
         $this->dataProvider = $dataProvider;
         $this->filter       = $filter;
     }
 
+    public function setCurrencyList(array $currencyList): self
+    {
+        $this->currencyList = $currencyList;
+
+        return $this;
+    }
+
+    public function setDefaultCurrency(CurrencyEntity $defaultCurrency): self
+    {
+        $this->defaultCurrency = $defaultCurrency;
+
+        return $this;
+    }
+
+    public function getCurrencyList(): array
+    {
+        return $this->currencyList;
+    }
+
+    public function getDefaultCurrency(): CurrencyEntity
+    {
+        return $this->defaultCurrency;
+    }
+
     public function generate(StreamInterface $stream, array $columns): void
     {
         $stream->addEntity($columns);
         $emptyRecord = array_combine($columns, array_fill(0, count($columns), ''));
         foreach ($this->dataProvider->getEntities() as $entity) {
-            $entityData = array_merge($emptyRecord, array_intersect_key($entity->toArray(), $emptyRecord));
+            $entityData = $this->calculatePrice(array_merge($emptyRecord, array_intersect_key($entity->toArray(), $emptyRecord)));
             $stream->addEntity($this->prepare($entityData));
         }
     }
 
+    private function calculatePrice(array $entityData): array
```

Comment:
I think it shouldn't be a  `Feed` responsibility to calculate the price. We have the field `Price` which is doing that. Please move that logic there